code/modules/fishing/sources/_fish_source.dm

```diff
@@ -132,7 +132,11 @@ GLOBAL_LIST_INIT(preset_fish_sources, init_subtypes_w_path_keys(/datum/fish_sour
 		return
 	if(isitem(reward)) //Try to put it in hand
 		INVOKE_ASYNC(fisherman, TYPE_PROC_REF(/mob, put_in_hands), reward)
-	fisherman.balloon_alert(fisherman, "caught [reward]!")
+		fisherman.balloon_alert(fisherman, "caught [reward]!")
+	else // for fishing things like corpses, move them to the turf of the fisherman
+		INVOKE_ASYNC(reward, TYPE_PROC_REF(/atom/movable, forceMove), get_turf(fisherman))
+		fisherman.balloon_alert(fisherman, "caught something!")
+
```

Comment:
I prefer it to be explicit, also because the reward is likely to appear partly hidden by the player mob at first, as most movables have a lower layer than mobs. But I won't press on.